Libraries/LibWebView/CookieJar.cpp

```diff
@@ -218,7 +221,7 @@ void CookieJar::expire_cookies_with_time_offset(AK::Duration offset)
 }
 
 // https://www.ietf.org/archive/id/draft-ietf-httpbis-rfc6265bis-15.html#section-5.1.2
-Optional<String> CookieJar::canonicalize_domain(const URL::URL& url)
+Optional<ByteBuffer> CookieJar::canonicalize_domain(const URL::URL& url)
```

Comment:
I may be misreading, but the spec seems to suggest that we should always be returning ASCII here

Reply:
I'm further confused, since I thought serialized_host would always return ascii as defined by the URL spec

Reply:
Yeah a lot of this pr's changes are to make it easier to use raw bytes in the places it actually matters.

Reply:
IMO if a function only returns ASCII or UTF-8, then we should use String, and we reserve ByteString for cases that it is actually a set of bytes. If the caller needs to combine this with some bytes, then they can reach into String::bytes()

Reply:
I'm also starting to think that this may be the entirely wrong approach since cookies should always be ASCII, these cookies should be rejected earlier in the process.

Reply:
Right, digging through I'm not quite sure I'm seeing the place where non ASCII is slipping through

Reply:
Well the path through `ResourceLoader::handle_network_response_headers > store_response_cookies > Cookie::parse_cookie` never checks that bytes are `< 0x80`

Reply:
Actually, [the latest Cookies spec](https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-rfc6265bis-17#name-the-set-cookie-header-field) says "the algorithm accommodates some characters that are not cookie-octets according to the grammar" and seemingly permits octets `>= 0x80`.

Reply:
Given this more permissive spec, only the cookie's domain-attribute is required to be ASCII. While I could change it back to being a `String`, I think that would make things more complicated since the code would need to handle both types for little benefit.

Reply:
Sure, the caller needs to handle it, but that seems fine, it's useful to see a String an be sure that it is valid unicode, ByteString means it won't be.

The primary point here though is that the FIXME being added here does not seem relevant - there is no non-ASCII host case to handle. It should always be ASCII.

On a related topic topic, I'm also a bit unsure how relevant the pre-existing FIXME actually is as well, may not actually be applicable. For example, we already do the ToASCII conversion that it mentions (part of URL), but that's a bit out of scope here.

Reply:
Ah, I understand what you're saying now. I suppose the two FIXMEs can be replaced with a "Note: This is handled by serialized_host"

Reply:
Right - and also that I don't understand the need to do an isomorphic encode here, I would personally prefer we keep our canonical internal representation of a domain as a String, consistent with the rest of the codebase. It's seems better to keep everything as a String where possible and convert at the API boundary if necessary rather than different encodings all over the place and needing to remember whether that ByteBuffer is latin-1, vs random bytes, vs whatever

Reply:
Yeah I'm in the process of changing the domain to be a String.